### PR TITLE
Fixes #245, where the automatic kernel selection yields incorrect values

### LINKED ISSUE #245: [Bug] Regression error in selecting correct automatic kernel size for instantaneous rate
**Describe the bug**
This is a potential regression introduced during development of the automatic kernel selection in #35, where a confusion between a kernel bandwidth and the standard deviation of the kernel was introduced. This led to non-optimal kernels in the instantaneous rate calculation as identified by @Junji110 .

Investigation and fix/regression test is in progress.

### PR DESCRIPTION
**Pull request by mdenker**
This fixes #245, adds a corresponding regression test, and clarifies a point regarding the output in the documentation of `instantaneous_rate()`

### CODE REVIEW

File: elephant/test/test_statistics.py [hunk 1]
```diff
@@ -570,6 +570,35 @@ def test_instantaneous_rate_regression_144(self):
         st = neo.SpikeTrain([2.12, 2.13, 2.15] * pq.s, t_stop=10 * pq.s)
         self.assertRaises(ValueError, es.instantaneous_rate, st, 1 * pq.ms)
 
+    # Regression test for #245
+    def test_instantaneous_rate_regression_245(self):
+        # This test makes sue that the correct kernel width is chosen when
```
Comment (Junji110) [inline elephant/test/test_statistics.py]: sue --> sure

File: elephant/test/test_statistics.py [hunk 2]
```diff
@@ -570,6 +570,35 @@ def test_instantaneous_rate_regression_144(self):
...
+        # selecting 'auto' as kernel
+        spiketrain = neo.SpikeTrain(
+            range(1, 30) * pq.ms, t_start=0*pq.ms, t_stop=30*pq.ms)
+
+        # This is the correct procedure to attain the kernel: first, the result
+        # of sskernel retrieves the kernel bandwidth of an optimal Gaussian
+        # kernel in terms of its standard deviation sigma, then uses this value
+        # directly in the function for creating the Gaussian kernel
+        kernel_width_sigma = es.sskernel(
+            spiketrain.magnitude, tin=None, bootstrap=True)['optw']
```
Comment (Junji110) [inline elephant/test/test_statistics.py]: Is `bootstrap=True` necessary?

Comment (mdenker) [reply elephant/test/test_statistics.py]: Indeed, it seems this could be left out.

File: elephant/statistics.py [hunk 3]
```diff
@@ -794,18 +795,13 @@ def instantaneous_rate(spiketrain, sampling_period, kernel='auto',
         raise ValueError("The sampling period must be larger than zero.")
 
     if kernel == 'auto':
-        kernel_width = sskernel(spiketrain.magnitude, tin=None,
-                                bootstrap=True)['optw']
-        if kernel_width is None:
+        kernel_width_sigma = sskernel(
+            spiketrain.magnitude, tin=None, bootstrap=True)['optw']
```
Comment (Junji110) [inline elephant/statistics.py]: The `bootstrap=True` here can also be left out.

**Comment by mdenker** [comment]
It seems the failing CI in 2.7 environment seems unrelated to this PR...

**Comment by dizcza** [comment]
I'm currently looking at this strange behavior with SSL error. I'll push a commit here once I find a solution.

**Pull request merged**